tests/unix_stream.rs

```diff
@@ -77,6 +77,55 @@ fn unix_stream_connect() {
     handle.join().unwrap();
 }
 
+#[test]
+fn unix_stream_connect_addr() {
+    let (mut poll, mut events) = init_with_poll();
+    let barrier = Arc::new(Barrier::new(2));
+    let local_addr = {
+        // Workaround through a temporary listener using the same address,
+        // as there is currently no way of directly building a `SocketAddr`.
+        let path = temp_file("unix_stream_connect_addr");
+        let listener = net::UnixListener::bind(path.clone()).unwrap();
+        let mio_listener = mio::net::UnixListener::from_std(listener);
+        let address = mio_listener.local_addr().unwrap();
+        drop(mio_listener);
+        _ = std::fs::remove_file(&path);
+        address
+    };
```

Comment:
Can't you just use `listener.local_addr()` after creating it?

Reply:
Absolutely yes, you are right. I pushed a commit on top to simplify this, and rebased everything to current `master`.